### ISSUE TITLE
Floating quantifiers

### ISSUE THREAD

**Issue by rafael75012** [UniversalDependencies/docs#581]
How should we annotate floating quantifiers?

I found many diverse strategies in actual English and French corpora 
In English LinES, I found 
(a) _they all go out_ : `advmod`(_go_, _all_)
(b) _they all had gone_ : `obl`(_gone_, _all_)

In EWT, I found 
(c) _we are all suffering_ : `advmod`(_suffering_, _all_)
(d) _you all remember_ : `det`(_you_, _all_)

In French_Sequoia : 
(e) _que nous connaissons tous_ : `advmod`(_connaissons_, _tous_)
(f) _qui seront tous deux mis en examen_ : `iobj`(_deux_, _mis_en_examen_)

In SpokenFrench : 
(g) _les constructeurs ne réagissent pas tous_ : `advmod`(_réagissent_, _tous_)
(h) _on est tous très inquiets_ : `obl:mod`(_inquiets_, _tous_)

In (e) and (g) the quantifier floats are tagged `ADJ`, but it might be a mistake because _tous_ here are pronouns I think

The (f) sentence relates to another problem : the delimitation of quantifier floats. 
In French_Sequoia :
(i) _d'autres chefs d'entreprise semblent, eux aussi ..._ : `iobj`(_semblent_, _eux_)

In fact "exclusivity markers" like these pronouns and quantifier floats have similar distributions (see Larrivée 2008): 
_Les caporaux ont_ [_tous_, _eux-mêmes_, _eux aussi_] _été renseignés_
_Les caporaux qui_ [_tous_, _eux-mêmes_, _eux aussi_] _avaient été renseignés_
_Il aurait fallu_ [_tous_, _eux-mêmes_, _eux aussi_] _qu'ils se renseignent_

Floating quantifiers behave like secondary predicates, they could be replaced by secondary predicate _seul_ 
_Les caporaux ont_ [_tous_, _seuls_] _été renseignés_
_Les caporaux qui_ [_tous_, _seuls_] _avaient été renseignés_
_Il aurait fallu_ [_tous_, _seuls_] _qu'ils se renseignent_

So I propose the following strategy : 
(e') _que nous connaissons tous_ : `acl`(_nous_, _tous_)
(i') _d'autres chefs d'entreprise semblent, eux aussi ..._ : `acl`(_chefs_, _eux_)

**Comment by arademaker** [comment]
@rafael75012 some of those cases are normal ambiguities, don't you think? So, maybe it is normal to do have different annotations.

**Comment by rafael75012** [comment]
Hi @arademaker, to what cases do you refer?

**Comment by arademaker** [comment]
Try examples (a) and (b) in a grammar like http://erg.delph-in.net/logon. You will see that both can have many different analyses.

In the IBM English Slot Grammar we got fewer analyses for (a) because of the parse prune mechanism:

```
They all go out.
------------------------------------------------------------------------
.- subj(n)  they(1)        noun pron pl def nom perspron
.- vadv     all3(2,3)      adv nounadv noadvpre nwh
o- top      go1(3,1,4,u,u) verb vfin vpres pl vsubj invertv vmove (nform going go-ahead) (ernform goer)
`- comp(pt) out1(4,3,u)    prep motionp badobjping
------------------------------------------------------------------------
------------------------------------------------------------------------
.- subj(n)  they(1)        noun pron pl def nom perspron
.- vadv     all3(2,3)      adv nounadv noadvpre nwh
o- top      go1(3,1,u,u,4) verb vfin vpres pl vsubj invertv vmove (nform going go-ahead) (ernform goer)
`- comp(lo) out1(4,3,u)    prep motionp badobjping
------------------------------------------------------------------------
```

**Comment by rafael75012** [comment]
If I am not wrong, http://erg.delph-in.net/logon, for "they all go out" and "they all had gone", propose two analyses of the quantifier float, attached to the pronoun "they" with a 'det' link, or attached to the verb with an 'adv' link.
Could you indicate me how "they--`det`-->all had gone" would differ semantically from "they all<--`advmod`--had gone"?

**Comment by arademaker** [comment]
Hi @rafael75012, this can be a long discussion about semantic representations. I recently read a very interesting article about it: https://aclweb.org/anthology/W/W15/W15-0128.pdf. The question is if these two different analyses are semantically identical and so they do not deserve to be considered ambiguities, right? 

I read the first one as an emphasis on the fact that the subject of the action is the whole set of people in the mind of the speaker. The second one is an emphasis on the completeness of the action.

**Comment by nschneid** [comment]
> I read the first one as an emphasis on the fact that the subject of the action is the whole set of people in the mind of the speaker. The second one is an emphasis on the completeness of the action.

I can only see the first reading. The use of "all" in that position is contingent on the subject being plural or mass. The following are terrible:

*John all had gone.
*John all has eaten the candies.

So I don't see an adverbial reading as a possibility here.

**Comment by arademaker** [comment]
@nschneid I am not a native speaker, but for both cases, given a singular subject, the adverbial reading was returned by the ERG/HPSG grammar.

**Comment by nschneid** [comment]
Then I think that indicates a problem with or limitation of the grammar.

(*That* sentence is definitely grammatical, by the way, but ERG rejects it.)

**Comment by arademaker** [comment]
@nschneid sorry, I didn't understand. ERG didn't reject `John all has eaten the candies.` nor `John all had gone.`.  If you are talking about the sentence 

> Then, I think that indicates a problem with or limitation of the grammar.

you are right, all grammars leak, right? But the version below was accepted.

> Then, I think that indicates a problem or limitation of the grammar.

I was just saying that grammars help us to see ambiguities that we missed sometimes. Moreover, making annotations on corpora force us to adopt one particular interpretation. Does it make sense?

**Comment by rafael75012** [comment]
Thank you @arademaker  and @nschneid  for your contributions
@arademaker , yes it was my question ! 

> The use of "all" in that position is contingent on the subject being plural or mass

Then wouldn't it be more accurate to use the treatment for depictives : `acl` when the verb is finite (linked to the argument) and `advcl` when the verb is non finite (linked to the verb)?

And there is also the question of the POS of these FQ

Would it be possible in English to remove the original subject, so that the role of subject would be taken by the FQ? If yes, it would be in favour of a `PRON` analysis?

> They all go out ==> ??All go out
> They all had gone ==> All had gone
> We are all suffering ==> All are suffering
> you all remember ==> ??All remember

In French, I am more inclined to tagg the FQ "tous" as a `PRON`, rather than `ADJ` as it has  been done in Sequoia_UD and FrenchSpoken

> que nous connaissons tous ==> que tous connaissons (a bit strange, but can be ok in some contexts)
> on est tous très inquiets ==> tous sommes très inquiets

**Comment by nschneid** [comment]
When "all" is used by itself as a subject, it feels like a quantifier/determiner being coerced into a pronoun. I think the policy is still to tag it as `DET`, but as `nsubj`.

When "all" is a floating quantifier, I think it makes sense to treat it as a dependent within the noun phrase—but probably not `acl`/`advcl`, as it is not a clause. EWT uses `det`, which seems reasonable:

- they all have the same unique grammar structure: `det(they, all)`

- the other five companies making the short list all proposed alternative structures: `det(companies, all)`

**Comment by sylvainkahane** [comment]
But we need first to decide whether the floating quantifier forms a phrase with the subject or with verb. It seems quite clear that it forms a phrase with the verb. For instance, it must be after the auxiliary:
<img width="341" alt="image" src="https://user-images.githubusercontent.com/23051796/47955085-fc8b3180-df92-11e8-8bc1-bb591699c1ec.png">
or can cooccur with a relative pronoun:
<img width="553" alt="image" src="https://user-images.githubusercontent.com/23051796/47955126-73282f00-df93-11e8-941d-03419219d04c.png">
So `advmod` seems a very good option, that has been adopted by most treebanks (even if we can find some inconstitencies sometimes).
The POS of _all_ is another story.

**Comment by nschneid** [comment]
No, it need not be after the auxiliary: "You all have won" and "You have all won" are both fine and semantically equivalent. Let's try VP coordination:

### Pre-aux

- They all have eaten breakfast but haven't eaten lunch.
- *They have eaten breakfast but all haven't eaten lunch. (only grammatical with the pronominal reading of "all", it which case it's not VP coordination)

Conclusion: if it precedes the auxiliary, this test suggests it's not part of the VP.

I think other adverbs can precede the aux, though it sounds a bit old-fashioned to my ears:

- They always have eaten lunch together but haven't played cards.
- They have eaten lunch together regularly and always have played cards.

The second one is no worse than the first, which distinguishes "always" from "all".

### Post-aux

- They have all eaten breakfast but haven't eaten lunch.
- They have eaten breakfast but haven't all eaten lunch.

Conclusion: this test doesn't rule out post-auxiliary "all" as part of the VP.

Incidentally, the following uses of "always" sound more natural than the pre-aux ones above:

- They have always eaten lunch together but haven't played cards.
- They have eaten lunch together regularly and have always played cards.

There are probably other tests that I'm not thinking of that could help.

**Comment by sylvainkahane** [comment]
In French, data are much simpler. The pronominal subject is clitic and the FQ cannot be between it and the verb: * _ils tous partent_, lit. they all leave vs _ils partent tous_ 'they all leave'
But we have a so-called strong form of the pronoun: _eux tous partent_ '(they all) leave' vs _eux partent tous_ 'they (all leave)'.
So it is likely that English has two constructions (with different prosodies?). But I guess that the default analysis is _they (all leave)_, which is the most most common in French. Just French intuition ;)

**Comment by rafael75012** [comment]
@nschneid , you said : 
> When "all" is a floating quantifier, I think it makes sense to treat it as a dependent within the noun phrase—but probably not acl/advcl, as it is not a clause

Do you mean that we cannot get :
> She entered the room sad : She entered the room + She was sad
> They all got out : They go out + *They were all

> EWT uses det, which seems reasonable

So it would be a "floating" version of determiners?
Determiners accompany NP normally, not pronouns...? 
FQ and determiners still differ a lot on these two properties

**Comment by nschneid** [comment]
Right, *They were all. doesn't work.

Certainly FQs are not typical determiners. But if one takes the view that "They all" is a noun phrase, I'm not sure there's a better relation than `det`.

**Comment by sylvainkahane** [comment]
In accordance with the [guidelines](http://universaldependencies.org/u/overview/morphology.html#pronominal-words), _all_ must be DET, whatever its position is. (The POS is an inherent value.)
But I'm not convinced by the relation `det` in _they all_. It is a very unusual position for a determiner in English. What is the commutation paradigm of _all_ in this position? In French, we can have _eux tous_ 'they all', _eux autres_ 'THEY', lil. 'they other', _eux seuls_ 'only they', lit. 'they alone'. Both _autres_ and _seuls_ are ADJs. So I think I would prefer to have `amod`.

**Comment by nschneid** [comment]
Here's what the Cambridge Grammar of the English Language (p. 413) has to say:

![image](https://user-images.githubusercontent.com/985263/48005844-7aaa1e00-e0e2-11e8-8100-ef19db2b257c.png)

I think it's just "all", "both", and "each" that can occur in this postmodifier construction (what they call *implicit partitives*).

**Comment by nschneid** [comment]
It's perhaps worth noting that "only" and "just" are tagged as `ADV` and accordingly attached as `advmod` when they modify nouns. So one could argue that even though these FQ constructions bear semantic and word order similarities between English and French, they deserve different dependency relations because they draw on different lexical categories.

(Or that the criteria for `DET`/`ADJ`/`ADV` are applied inconsistently across languages.)

**Comment by rafael75012** [comment]
@nschneid, would you annotate "explicit partitives" and "implicit partitives" in the same way? With `det` relation and `DET` pos? 

@sylvainkahane 
> What is the commutation paradigm of _all_ in this position? In French, we can have eux tous 'they all', eux autres 'THEY', lil. 'they other', eux seuls 'only they', lit. 'they alone'

In French, do we have an equivalent to English PRON + all (i.e. "they all")? 

> (1) *Ils tous sortent
> they + all + go out
> (2) *Ils chacun sortent
> they + both + go out
> 

If not, then "eux aussi", etc. commute with "tous" solely in the two following positions : (3) placed after in between AUX + VERB, (4) placed after the verb

> 
> (3) Ils sont [tous, eux-mêmes, eux aussi] sortis
>      they + AUX + ['all', 'themselves', 'them too'] + gone out
> 
> (4) Ils sortent [tous, eux-mêmes, eux aussi]
>       they + go out + ['all', 'themselves', 'them too']
> 

French FQ would be syntactically equivalent to English "expliticit partitives" mentioned in the Cambridge Grammar, but not with "implicit partitives"

Note that the inclusion of FQ is marked prosodically with an emphasis (which can be transcribed with commas), and we also have the dislocation alternation : 
> (5) [Tous, eux-mêmes, eux aussi], ils sont  sortis
>      ['all', 'themselves', 'them too'] + they + AUX + gone out
> 
> (6) Ils sortent, [tous, eux-mêmes, eux aussi]
>       they + go out + ['all', 'themselves', 'them too']
> 
Using `dislocated` could also make sense...

**Comment by rafael75012** [comment]
For FQ in French, could we use `acl`? 
Once we admit that they behavior is similar to secondary predicates, and that `advcl` has been proposed to replace `acl` for their description (*https://github.com/UniversalDependencies/docs/issues/476), `advcl` for FQ could be an option to

**Comment by dan-zeman** [comment]
I am inclined to use `det`, as proposed by @nschneid above. It may be unusual from the point of view of traditional grammar, but it seems in line with how quantifiers are treated in UD.

**Comment by sylvainkahane** [comment]
`det` is a very particular relation that we need  for languages such as English were a special word is the functional head of the noun phrase (and cannot in particular be easily erased). This construction is so special that I would be very reluctant to use it for something else.

For FQ in French, we have two constructions: one where the FQ forms a phrase with the verb (_ils sont tous venus_ 'they all came', lit. they have all come) and one, more marginal, where the FQ forms a phrase with a (strong form of a) pronoun (_eux tous sont venus_ 'all of them came', lit. they all have come). It seems simpler to use `advmod` when the FQ depends on a verb and `amod` when it depends on a pronoun.

**Comment by rafael75012** [comment]
@dan-zeman Even for French, in which we find FQ made of a two tokens? And in which the placement is more free, and in which they are not adjacent to the element they are supposed to modify?
> 
> Ils sortent, [tous, eux-mêmes, eux aussi]
> they + go out + ['all', 'themselves', 'them too']
> 

Just note in passing that FQ in English are very similar to FQ in Chinese, so if `DET` is adopted in English, it may also fit for Chinese : 

> (1) 他们 都 走 了
>  tāmen + dōu +  zǒu + le
> they + all + go + PFV
> 
> (2) 他们 俩 走 了
> Tāmen + liǎ + zǒu +le
> they + two + go + PFV

In (2) 俩 is a numeral.

**Comment by dan-zeman** [comment]
@rafael75012 : Yes, why not. There is no universal rule that a modifier must be adjacent to the element it modifies. Or that it must consist of a single token.

In your Chinese examples, I would do
`det(tāmen, dōu)`
and
`nummod(tāmen, liǎ)`,
respectively.

@sylvainkahane : Maybe the `det` relation in UD is not as specific as outside UD, but it has a misleading name. I just had a look at the guidelines and the [det page](https://universaldependencies.org/u/dep/det.html) does not mention quantifiers but the [nummod page](https://universaldependencies.org/u/dep/nummod.html) says that indefinite quantifiers should be attached via `det`.

I find it difficult to accept that _tous_ forms a phrase with the verb in _ils sont tous venus_. Why couldn't _ils tous_ be a (discontinuous) phrase? And isn't there some sort of agreement in number and gender? I suppose _*il est tous venu_ is ungrammatical, no?

**Comment by rafael75012** [comment]
@dan-zeman About the Chinese example, do you refer to this rule "However, cardinal numerals in the narrow sense (one, five, hundred) are not tagged `DET` even though some authors would include them in quantifiers. Cardinal numbers have their own tag `NUM`."? https://universaldependencies.org/u/pos/DET.html

So it would be like in the French

> Ils sont tous (les) deux venus

Where you would get : 

> nummod (ils, deux/NUM)
> det (deux/NUM, tous/DET)
> det (deux/NUM, les/DET)

By the way, do we agree that "tous/toutes" when they are not substantival pronouns, are `DET` and not `ADJ`? 
> 
> tous les chats sont gris ('all cats are grey')
> toutes les fenêtres sont cassées ('all windows are broken')

**Comment by dan-zeman** [comment]
_tous_ `DET` is fine with me. Accompanied by `PronType=Tot`.

**Comment by sylvainkahane** [comment]
@dan-zeman Agreement between two units does not mean that they form a phrase together (cf. anaphoric relations). I don't see any serious argument to consider that the FQ forms a phrase with the subject. French weak subject pronouns such as _ils_ are clitics and cannot have a dependent. They can only be separated from the verb by other clitics.

**Comment by rafael75012** [comment]
@dan-zeman It agrees with the subject/object like a depictive would agree with it

> Ils sont rentrés tristes | Ils sont tous rentrés
> Il est rentré triste | Elles sont toutes rentrées
> Il les a vues tristes | Ils les a toutes vues
> 

So if there is no "any serious argument to consider that the FQ forms a phrase with the subject" (@sylvainkahane), except if there are arguments to consider that FQ forms a phrase with the verb, `acl` would be ok I think, as FQ relation is neither fully with the verb neither with its argument

**Comment by wongtaksum** [comment]
倆 liǎr = two pieces
I do not think it is a numeral while 倆liar refers to 他們tamen.  I would use appos(倆, 他們).
What is the problem of treating 都dou as an adverb?

**Comment by dan-zeman** [comment]
Note: This issue seems to overlap with older #484.

**Comment by Stormur** [comment]
No one has mentioned `xcomp`? This seems a clear case to me, as it looks like a secondary predication of the subject.

They are all gone = They are gone, and "_those who went were all_" 

Something along these lines. Of course the `DET` always stays `DET` and, in languages with richer morphology, it  concords with the subject. It is by no means different than the famous Latin sentence:

_Galli **laeti** in castra pergunt_ 'The Gauls **gladly** enter the camp'

One could unproblematically substitute _laeti_ 'glad (masculine plural nominative)' with _omnes_ 'all':

_Galli **omnes** in castra pergunt_ 'The Gauls **all** enter the camp', 'The Gauls enter the camp _**in their totality**_'

On a semantic level, it is not _all the Gauls_: it is just _all those Gauls_ at hand.

It is similar to a depictive and it is core inasmuch as it involves the subject. By the way, hoping not to be off-topic, I exploit this occasion to comment that the final line in the guidelines about the `xcomp` deprel regarding the use of `acl` instead of `xcomp` in such occasions does not seem acceptable to me as it stands now: on the one hand, it makes a point using a complement/adjunct logic (_leaving it out will neither affect grammaticality nor significantly alter the meaning of the verb_) alien to UD, and on the other hand it would lead to an analysis that, at least in Latin (especially given its freer word order), is simply wrong (it is not the _happy Gauls_ that enter the camp, but the Gauls that _enter gladly_).

**Comment by nschneid** [comment]
> I exploit this occasion to comment that the final line in the guidelines about the `xcomp` deprel regarding the use of `acl` instead of `xcomp` in such occasions does not seem acceptable to me as it stands now: on the one hand, it makes a point using a complement/adjunct logic (_leaving it out will neither affect grammaticality nor significantly alter the meaning of the verb_) alien to UD, and on the other hand it would lead to an analysis that, at least in Latin (especially given its freer word order), is simply wrong (it is not the _happy Gauls_ that enter the camp, but the Gauls that _enter gladly_).

I recently bumped up against this issue for English, where I expected depictives to be `xcomp` and was surprised to find the guidelines specify `acl`. (Maybe I missed or forgot the discussion that led to this.) In #476 there seems to be some support for moving away from `acl`.

**Comment by dan-zeman** [comment]
> (Maybe I missed or forgot the discussion that led to this.)

This may be baggage from the Stanford Dependencies. [The section](https://universaldependencies.org/docsv1/u/overview/specific-syntax.html#secondary-predicates) was already in the v1 guidelines, added by @manning on 8 October 2014 (e5e3018045cca7ad5bd6b9f52a6e7d270aa8d3de), following e-mail-based discussion (we were not using Github issues yet). I admit that it has been criticised repeatedly because of the reference to obligatoriness/optionality, which is otherwise taboo in UD. I'm afraid we have to live with it for now (and maybe forever), as changing the guideline would mean that the phenomenon has to be identified and re-annotated in all treebanks.
